.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,50 @@
+name: "Code scanning - scheduled (weekly) or on-demand"
+
+on:
+  schedule:
```

Comment:
Not sure if the v2 branch can be scanned with the schedule option, it didn't seem to have it from the docs.

Reply:
Re. `push`, we purposely are not enabling those yet, because we want to see results/value & time-to-scan via async scheduled/dispatch runs before we start loading existing GitHub use (PRs) up with longer-running checks (`push`, `pull_request`, etc).

Re. the `schedule` option, yes - from looking at the docs, it seems that'll only get the default branch.  We'll be able to trigger scans of the `v2` branch manually for now via the `workflow_dispatch` setup which lets  you specify the branch to run against in the Actions UI.

I'd like to get this merged for now, so we can see how the results & time-to-scan look, then we can see about tweaking the config for better coverage.